### ISSUE TITLE
new_folder_permissions not respected for cache folder(s)

### ISSUE THREAD

**Issue by wuuti** [modxcms/revolution#13586]
### Summary
The `new_folder_permissions` permission is not respected for all cache folders.

### Step to reproduce
Set the system setting `new_folder_permissions` to a value, e.g. `0700`. Delete the cache folder in the filesystem. Reload a manager or frontend page. The cache folder and its contents are regenerated, but with strange permissions.

### Observed behavior
```bash
drwxr-xr-x 17 user psacln 4096 Sep  2 19:29 ./
drwxr-xr-x 14 user psacln 4096 Sep  2 19:28 ../
drwx------  3 user psacln 4096 Sep  2 19:29 context_settings/
drwx------  3 user psacln 4096 Sep  2 19:29 db/
drwxr-xr-x  3 user psacln 4096 Sep  2 19:29 default/
drwx------  3 user psacln 4096 Sep  2 19:29 includes/
drwx------  3 user psacln 4096 Sep  2 19:29 lexicon_topics/
drwx------  2 user psacln 4096 Sep  2 19:29 logs/
drwx------  3 user psacln 4096 Sep  2 19:29 media_sources/
drwx------  2 user psacln 4096 Sep  2 19:29 namespaces/
drwx------  3 user psacln 4096 Sep  2 19:29 resource/
drwx------  3 user psacln 4096 Sep  2 19:29 scripts/
drwxr-xr-x  2 user psacln 4096 Sep  2 19:29 system_settings/
```
### Expected behavior
Every folder, including the "root" cache folder, should be generated with the desired 0700 (which is drwx------) permissions. Instead, the cache folder itself, the default folder and the system_settings folder actually get a 0755. 

### Environment
Tested with MODX 2.5.7, PHP 5.5.9, Apache 2.4.7

**Comment by wuuti** [comment]
Additionally, first quick code review revealed also that on several places in the modx core different default permission values are set:

* core/model/modx/processors/system/filesys/file and /folder: 0777
* manager/controllers/default/workspaces: 0775

Despite the fact, that the default values should be the same over the installation, shouldn't those values be the most restrictive possible, say at best 0644 for files?

**Comment by wuuti** [comment]
Can anyone confirm that?

**Comment by Mark-H** [comment]
This is fixed in 2.6. See #12677 and modxcms/xpdo#89.
